### DISCUSSION TITLE
generate的拓展方法，应该写在哪个文件才会生效呢？

### DISCUSSION THREAD

**Discussion by csywweb** [umijs/umi#11493] [Q&A]
<img width="939" alt="image" src="https://github.com/umijs/umi/assets/3185567/30c1484e-d2ca-49e8-9c34-07115c875fc6">
文档中的代码看懂了，不知道写在哪里

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4

**Comment by csywweb** [comment]
那umi4 能自定义微生成器吗

**Reply by fz6m** [reply]
可以通过编写 umi 插件或者 [项目级插件](https://umijs.org/docs/guides/use-plugins#%E9%A1%B9%E7%9B%AE%E7%BA%A7%E6%8F%92%E4%BB%B6) 来注册自己的生成器，代码可参考 https://github.com/umijs/umi/tree/master/packages/preset-umi/src/commands/generators 。
